### DISCUSSION TITLE
Solana usage with web3 modal

### DISCUSSION THREAD

**Discussion by Cliv3** [WalletConnect/walletconnect-monorepo#3664] [Developer Support] [answered]
https://www.youtube.com/watch?v=iAuLOaSOFSM 

Hi guys I came across this video and I want to implement something similar and connect wallet connect using Solana. Any pointers? the solib library doesn't seem to work or is not very Clear. Any pointers?

**Comment by boidushya** [comment, marked as answer]
Hey @Cliv3 !
[Here's our Solana implementation example](https://github.com/WalletConnect/web-examples/tree/main/basic/dapps/universal-provider-solana)
Alternatively, you can also check out live demos on codesandbox with 
- [Sign Client](https://codesandbox.io/p/sandbox/walletconnect-solana-sign-client-vrjlsz?file=/src/App.tsx:1,1)
- [Universal Provider](https://codesandbox.io/p/sandbox/wallectconnect-solana-universal-provider-czlm72?file=/src/App.tsx:1,1)

**Reply by Cliv3** [reply]
Wow, Thank you very much. And just to clarify what is the difference between sign client and unviversal provider?

**Reply by boidushya** [reply]
So, the provider allows RPC connection to blockchain nodes, whereas sign-client handles connection between dapps and wallets
